### ISSUE TITLE
SBOL 2-3 converter - Component (Priority 2)

### ISSUE THREAD

**Issue by PrashantVaidyanathan** [SynBioDex/SBOL-utilities#246]

**Comment by Yehuda-Binik** [comment]
@jakebeal 

Me and @PrashantVaidyanathan were discussing this issue. It appears that `visit_component` has not yet been completed. Currently, the function implements correctly the following line on page 51 from the SBOL3 documentation: "SBOL 2.x `ComponentDefinition` objects map to SBOL 3.x `Component` objects. The type property is mapped according to Table 19. " However, the function does not implement the following line in the documentation: "SBOL 2.x `ModuleDefinition` objects map to SBOL 3.x `Component` objects with a type of SBO:0000241 (func- 18tional entity)." How should the function be modified to take this into account?

**Comment by jakebeal** [comment]
Yes, it has not yet been implemented and needs to be. Not sure how much more to say than that...

**Comment by bbartley** [comment]
An SBOL3 Component may actually map into SBOL2 as _both_ a `ComponentDefinition` and a `ModuleDefinition`.  Currently the `SBOL3To2ConversionVisitor.visit_component` method only converts to one or the other.  Leaving a note here that:

1. If there are any `interactions`, `interface`, or `models`, or SBO function entity type, then there must by a `ModuleDefinition`
2. If there are any `constraints`, `sequences`, or `features` with a `Location`, then there must be a `ComponentDefinition`.
